packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -340,16 +340,24 @@ export function calculatePositionInternal(
 
   let arrowPosition: Position = {};
 
-  const preferredArrowPosition = overlaySize[crossSize] / 2;
-
-  const arrowOverlappingChildMinEdge = childOffset[crossAxis] - overlaySize[crossAxis] + (arrowCrossSize / 2);
-  const arrowOverlappingChildMaxEdge = childOffset[crossAxis] + childOffset[crossSize] - overlaySize[crossAxis] - (arrowCrossSize / 2);
+  // All coords are transformed so that (0, 0) is at the top left of the overlay
+  // Prefer the arrow being in the center of the trigger/overlay anchor element
+  let preferredArrowPosition = childOffset[crossAxis] + .5 * childOffset[crossSize] - overlaySize[crossAxis];
 
+  // Min/Max position limits for the arrow with respect to the overlay
   const arrowMinPosition = arrowCrossSize / 2;
   const arrowMaxPosition = overlaySize[crossSize] - (arrowCrossSize / 2);
 
-  const arrowPositionOverlappingChild = clamp(preferredArrowPosition, arrowOverlappingChildMinEdge, arrowOverlappingChildMaxEdge);
+  if (preferredArrowPosition < arrowMinPosition || preferredArrowPosition > arrowMaxPosition) {
```

Comment:
this should get taken care of in the clamp?

Reply:
ah, I actually can just skip the clamps entirely if the center of the button is within the arrowMinPosition/maxPosition. The clamps are more for attempting to position the arrow in the center of the overlay or as close as possible if it exceeds the bounds. I'll refactor a bit